scripts/helpers/todos.js

```diff
@@ -0,0 +1,124 @@
+const TypeScript = require("typescript");
+const { isFile, readFile, removeFile, writeFile } = require("./file-system");
+const { getBranch } = require("./git");
+const { getLineAtOffset, parseLines } = require("./text");
+
+const { formattedDateTime } = require("./time");
+const { Project } = require("../helpers/project");
+
+const name = "TODOS.md";
+
+/**
+ * @typedef {Object} Line
+ * @property {string} value
+ * @property {number} index
+ * @property {number} start
+ * @property {number} end
+ */
+
+/**
+ * @typedef {Object} AnnotadedComment
+ * @property {string} file
+ * @property {TypeScript.TodoComment} comment
+ */
+
+/**
+ * @type {Array<TypeScript.TodoCommentDescriptor>}
+ */
+const checks = [
+  { text: "@todo", priority: 1 },
+  { text: "@hack", priority: 1 },
+  { text: "@bug", priority: 1 },
+  { text: "@fixme", priority: 1 }
+];
+
+// const lines = /**@type {Set<String>} */ (new Set());
+
+/**
+ * @param {string} file
+ * @param {Project} project
+ * @return {Array<AnnotadedComment>}
+ */
+function computeComments(file, project) {
+  /**
+   * @type {Array<AnnotadedComment>}
+   */
+  const comments = [];
+  for (const comment of project.getTodos(file, checks)) {
+    comments.push({ file: file, comment: comment });
+  }
+
+  return comments;
+}
+
+/**
+ * @param {Array<AnnotadedComment>} todos
```

Comment:
Index the todos by file to simplify things: `Map<string, TypeScript.TodoComment>`. It alleviates the need for a "cache" of parsed lines among other things.